### ISSUE TITLE
802.11S ??? WORK?

### ISSUE THREAD

**Issue by vigeruno** [barbieri/barbieri-playground#5]
I tried your scrypt and it works visually but the STAs really don't get disconnected as ever?

The test was performed in MESH 802.11S mode

wifi-disconnect-low-signal: wlan0: 90:F6:52:40:03:8E signal=-68dBm snr=26dBm strikes=3 (threshold: snr_stay=27 strikes=3): should disconnect
wifi-disconnect-low-signal: wlan0: disconnected 90:F6:52:40:03:8E {signal=-68dBm, noise=-94dBm, snr=26dBm, strikes=3, is_new=false}
wifi-disconnect-low-signal: wlan0: signal poll: 1 STAs, 1 disconnected
wifi-disconnect-low-signal: wlan0: signal poll: 0 STAs, 0 disconnected
wifi-disconnect-low-signal: wlan0: signal poll: 0 STAs, 0 disconnected
wifi-disconnect-low-signal: wlan0: signal poll: 0 STAs, 0 disconnected
wifi-disconnect-low-signal: wlan0: signal poll: 0 STAs, 0 disconnected
wifi-disconnect-low-signal: wlan0: signal poll: 0 STAs, 0 disconnected
wifi-disconnect-low-signal: wlan0: signal poll: 0 STAs, 0 disconnected
wifi-disconnect-low-signal: wlan0: signal poll: 0 STAs, 0 disconnected
wifi-disconnect-low-signal: wlan0: signal poll: 0 STAs, 0 disconnected
wifi-disconnect-low-signal: wlan0: signal poll: 0 STAs, 0 disconnected
wifi-disconnect-low-signal: wlan0: signal poll: 0 STAs, 0 disconnected
wifi-disconnect-low-signal: wlan0: signal poll: 0 STAs, 0 disconnected
wifi-disconnect-low-signal: wlan0: signal poll: 0 STAs, 0 disconnected
wifi-disconnect-low-signal: wlan0: forget 90:F6:52:40:03:8E
wifi-disconnect-low-signal: wlan0: 90:F6:52:40:03:8E signal=-55dBm snr=40dBm strikes=0 (threshold: snr_connect=33 strikes=0): good signal
wifi-disconnect-low-signal: wlan0: signal poll: 1 STAs, 0 disconnected
wifi-disconnect-low-signal: wlan0: 90:F6:52:40:03:8E signal=-43dBm snr=52dBm strikes=0 (threshold: snr_stay=27 strikes=3): good signal
wifi-disconnect-low-signal: wlan0: signal poll: 1 STAs, 0 disconnected
wifi-disconnect-low-signal: wlan0: 90:F6:52:40:03:8E signal=-42dBm snr=53dBm strikes=0 (threshold: snr_stay=27 strikes=3): good signal
wifi-disconnect-low-signal: wlan0: signal poll: 1 STAs, 0 disconnected
wifi-disconnect-low-signal: wlan0: 90:F6:52:40:03:8E signal=-42dBm snr=53dBm strikes=0 (threshold: snr_stay=27 strikes=3): good signal
wifi-disconnect-low-signal: wlan0: signal poll: 1 STAs, 0 disconnected

**Comment by barbieri** [comment]
it should have disconnected:

```
wifi-disconnect-low-signal: wlan0: disconnected 90:F6:52:40:03:8E {signal=-68dBm, noise=-94dBm, snr=26dBm, strikes=3, is_new=false}
```

at least this does the command to, when I first wrote this it was to force my phone to roam to another AP and it did work.

**Comment by jxph1123** [comment]
In fact, signal_connect and signal_snr_connect are useless, they will not prohibit wifi connection, and will repeatedly connect and remove them.

**Comment by barbieri** [comment]
yes, there is no way (AFAIK) to prohibit the connection, just force the disconnect.

The reasoning is that most wifi devices will try to keep connected to the same AP as much as possible, but if you roam in your house, the signal may become weak, which will require more retransmissions and make the quality worse... and it may be that there is a much better AP nearby, but the wifi won't "jump" to the new AP unless you kick it from the previous AP (or it becomes disconnected due some other reason), then it will look for the best signal with the same SSID.

**Comment by jxph1123** [comment]
In fact, the WIFI weak signal disconnect function has been implemented, which is very useful for AP switching at home. Thank you for the excellent tool.

**Comment by barbieri** [comment]
could we close this?
